pkg/action/custom_resource_lister.go

```diff
@@ -0,0 +1,138 @@
+		if resource.Kind == v1alpha1.ClusterServiceVersionKind {
+			csvKey.Name = resource.Name
+			csvKey.Namespace = resource.Namespace
+			break
+		}
+	}
+
+	if csvKey.Name == "" && csvKey.Namespace == "" {
+		return nil, fmt.Errorf("could not find underlying CSV for operator %s", operator.Name)
+	}
+
+	err := client.Get(ctx, csvKey, &csv)
+	if err != nil {
+		return nil, fmt.Errorf("could not get %s CSV on cluster: %s", csvKey.String(), err)
+	}
+
+	// check if owned CRDs are defined on the csv
+	if len(csv.Spec.CustomResourceDefinitions.Owned) == 0 {
+		return nil, fmt.Errorf("no owned CustomResourceDefinitions specified on CSV %s, no custom resources to display", csvKey.String())
+	}
+
+	return csv.Spec.CustomResourceDefinitions.Owned, nil
+}
+
+// List takes in a CRD description and finds the associated CRs on-cluster.
+// List can return a potentially unbounded list that callers may need to paginate.
+func List(ctx context.Context, crClient client.Client, crdDesc v1alpha1.CRDDescription, namespace string) (*unstructured.UnstructuredList, error) {
+	result := &unstructured.UnstructuredList{}
+
+	// find CRD on-cluster to determine CRD scope (not included in description)
+	crd := apiextensionsv1.CustomResourceDefinition{}
+	crdKey := types.NamespacedName{
+		Name: crdDesc.Name,
+	}
+	err := crClient.Get(ctx, crdKey, &crd)
+	if err != nil {
+		return nil, nil
+	}
+	scope := crd.Spec.Scope
+
+	if scope == apiextensionsv1.ClusterScoped {
+		// get all CRs across the cluster for the given CRD since namespace is not relevant for cluster-scoped CRs
+		result := unstructured.UnstructuredList{}
+		gvk := schema.GroupVersionKind{
+			Group:   crdDesc.Name,
+			Version: crdDesc.Version,
+			Kind:    crdDesc.Kind,
+		}
+		result.SetGroupVersionKind(gvk)
+		if err := crClient.List(ctx, &result); err != nil {
+			return nil, err
+		}
+		return &result, nil
+
+	} else if scope == apiextensionsv1.NamespaceScoped {
+		// get CRs in the cluster for the given namespace only
+		result := unstructured.UnstructuredList{}
+		gvk := schema.GroupVersionKind{
+			Group:   crdDesc.Name,
+			Version: crdDesc.Version,
+			Kind:    crdDesc.Kind,
+		}
+		result.SetGroupVersionKind(gvk)
+
+		options := client.ListOptions{Namespace: namespace}
+		if err := crClient.List(ctx, &result, &options); err != nil {
+			return nil, err
+		}
+		return &result, nil
+	}
+
+	return result, nil
+}
+
+// ListAll wraps the above functions to provide a convenient command to go from package/namespace to custom resources.
+func ListAll(ctx context.Context, client client.Client, opKey types.NamespacedName) (*unstructured.UnstructuredList, error) {
+	operator, err := FindOperator(ctx, client, opKey)
+	if err != nil {
+		return nil, err
+	}
+
+	crdDescs, err := Unzip(ctx, client, operator)
+	if err != nil {
+		return nil, err
+	}
+
+	var result unstructured.UnstructuredList
+	for _, crd := range crdDescs {
+		list, err := List(ctx, client, crd, opKey.Namespace)
```

Comment:
I think we want to search all the target namspaces as defined by the operator group in `opKey.Namespace`, right?

If so, my suggestion would be to:
1. List operator groups (make sure there's exactly 1)
2. Extract `status.Namespaces` from the operator group
3. List operands across the cluster
4. Filter the results from 3 based on the operator group's target namespaces.

If you do it this way, I don't think you'll have to worry about CRD scope since you can just always include items where `item.GetNamespace() == ""`